### ISSUE TITLE
Etcd file in local etcd filesystem is not getting removed after deleting from UI

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#24775]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create custom cluster (1 Control Plane/ 1 etcd / 3 workers) I used EC2 instances.
2. Make a new snapshot after initial backup.
3. Delete the new snapshot. This reference gets deleted from the Rancher backend.
4. Login in the cluster's etcd local filesystem and look for contents of `/opt/rke/etcd-snapshots/`

**Result:**

The snapshots files created and then deleted from the rancher UI are still persisting in the etcd's filesystem.

**Other details that may be helpful:**

`Ubuntu 18.04.3 LTS 4.15.0-1051-aws`

**Environment information**
- Rancher version: **2.3.4-rc5**
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): aws cloud
- Kubernetes version (use `kubectl version`):

```
v1.16.4
```

- Docker version (use `docker version`):

```
Docker version 19.03.5
```

**Comment by soumyalj** [comment]
Reproduced the issue with the same steps above in v2.3.3.  None of the snapshots get deleted from `/opt/rke/etcd-snapshots/` directory of the etcd node after deleting from the UI

**Comment by superseb** [comment]
This is fixed in https://github.com/rancher/rancher/pull/24398 and needs a backport.

**Comment by superseb** [comment]
@izaac Can you verify it works on `master-head`? Then https://github.com/rancher/rancher/pull/24817 should fix this for 2.3.x

**Comment by soumyalj** [comment]
The issue is verified in master-head. The snapshots deleted from Rancher UI, get deleted successfully from /opt/rke/etcd-snapshots directory of the etcd node.

**Comment by soumyalj** [comment]
Tested with v2.3.4-rc6
The issue is fixed in v2.3.4-rc6. The snapshots deleted from Rancher UI get deleted successfully from /opt/rke/etcd-snapshots directory of the etcd node.
Before deletion:
```
ubuntu@ip-172-31-1-189:/opt/rke/etcd-snapshots$ ls -ltr
total 1036
-rw------- 1 root root 329446 Jan  2 23:58 c-dfds5-rl-qfcr9_2020-01-02T23:58:31Z.zip
-rw------- 1 root root 361514 Jan  3 00:03 c-dfds5-ml-jkll2_2020-01-03T00:03:05Z.zip
-rw------- 1 root root 362350 Jan  3 00:03 c-dfds5-ml-gt4tv_2020-01-03T00:03:10Z.zip

```
After deletion:
```
ubuntu@ip-172-31-1-189:/opt/rke/etcd-snapshots$ ls
ubuntu@ip-172-31-1-189:/opt/rke/etcd-snapshots$ 
```
